src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5855,6 +5867,208 @@ private BoundExpression BindTypeParameterCreationExpression(ObjectCreationExpres
             return MakeBadExpressionForObjectCreation(node, typeParameter, analyzedArguments, initializerOpt, typeSyntax, diagnostics);
         }
 
+        private BoundExpression BindCollectionLiteralExpression(CollectionCreationExpressionSyntax syntax, TypeSymbol targetType, BindingDiagnosticBag diagnostics)
+        {
+            // Follows approach from BindCollectionInitializerExpression.
+
+            // PROTOTYPE: Test types other than NamedTypeSymbols such as arrays and type parameters.
+            // PROTOTYPE: Test constructor with all optional parameters; with params parameter.
+            var constructor = ((NamedTypeSymbol)targetType).Constructors.SingleOrDefault(c => c.ParameterCount == 0);
```

Comment:
note: i think it's fine for prototype to only do NamedTypeSymbol.  however, i worry aobut this *crashing* if it's another type.  Maybe better to do a type check return BadExpression if it's not a named type?